app/tests/data_factory.py

```diff
@@ -0,0 +1,978 @@
+        if links:
+            json_data['links'] = [link.json for link in self.attr_links]
+        return json_data
+
+    @property
+    def model_providers(self):
+        '''Returns a django DB model object of the providers sample data
+
+        If the data has not yet been created in DB, then it is created.
+
+        Returns:
+            list of provider model instances
+        '''
+        if not self.model_providers_instance:
+            attributes = self.get_attributes(remove_relations=False)
+            providers = attributes.pop('providers', [])
+            if not self.model_instance:
+                self._create_model(attributes)
+            self._create_model_providers(providers)
+        return self.model_providers_instance
+
+    @property
+    def model_links(self):
+        '''Returns a django DB model object of the links sample data
+
+        If the data has not yet been created in DB, then it is created.
+
+        Returns:
+            list of link model instances
+        '''
+        if not self.model_links_instance:
+            attributes = self.get_attributes(remove_relations=False)
+            links = attributes.pop('links', [])
+            if not self.model_instance:
+                self._create_model(attributes)
+            self._create_model_links(links)
+        return self.model_links_instance
+
+    def _create_model(self, attributes):
+        attributes.pop('providers', None)
+        attributes.pop('links', None)
+        self.model_instance = Collection(**attributes)
+        self.model_instance.full_clean()
+        self.model_instance.save()
+
+    def _create_model_providers(self, providers):
+        for provider in providers:
+            provider_instance = Provider(collection=self.model_instance, **provider)
+            provider_instance.full_clean()
+            provider_instance.save()
+            self.model_providers_instance.append(provider_instance)
+
+    def _create_model_links(self, links):
+        for link in links:
+            link_instance = CollectionLink(collection=self.model_instance, **link)
+            link_instance.full_clean()
+            link_instance.save()
+            self.model_links_instance.append(link_instance)
+
+
+class ItemLinkSample(LinkSample):
+    '''Item's Link Sample Data
+    '''
+    model_class = ItemLink
+    sample_name = 'item-link'
+    samples_dict = item_link_samples
+
+
+class ItemSample(SampleData):
+    '''Item Sample Data
+    '''
+    model_class = Item
+    sample_name = 'item'
+    samples_dict = item_samples
+    key_mapping = {'name': 'id'}
+    optional_fields = ['properties_title', 'links']
+
+    def __init__(self, sample='item-1', collection=None, name=None, **kwargs):
+        '''Create a item sample data
+
+        Args:
+            sample: string
+                Name of the sample based to use, see tests.sample_data.item_samples
+            required_only: bool
```

Comment:
`required_only` is not an argument of `__init__`, probably docstring has not been updated after changes?